Middleware 1.3.16 release notes (#31)
fiskaltrust/release-notes: middleware/middleware-1.3.16.md

---
slug: /release-notes/middleware/1.3.16
title: Version 1.3.16
---

# fiskaltrust.Middleware 1.3.16 (Germany)
_March 12, 2021_

> We're happy to announce that the fiskaly SCU is now available in our production systems (please see our [blog post](https://fiskaltrust.de/news/aktueller-status-zur-cloud-tse-202103/) for more details). Additionally, we've added a new signature that contains the TSE certification ID and (if required) information about the state of the certification and operational environment regulations. Finally, this version contains some bug fixes and usability improvements for the Middleware.

Reviewer: See comments from above

<div class="alert alert--warning" role="alert">Version 1.3 of the Middleware is meant for the German market only, customers in Austria and France should continue to use version 1.2. We will unify these experiences in an upcoming version.</div>

## Production release: fiskaly SCU

> In close collaboration with our partners, we've released the fiskaly SCU to production. To fulfill the legal requirements, we followed fiskaly's advice and added the informational phrase _"TSE in Evaluierung"_ ("TSE in evaluation") into the newly added certification identification signature item (see below). Please keep in mind that the fiskaly TSE is still in evaluation, and switching to the certified state will most likely require an update of the SCU. Please refer to our [blog post about the state of cloud TSEs](https://fiskaltrust.de/news/aktueller-status-zur-cloud-tse-202103/) for more details.

Reviewer: ```suggestion
In close collaboration with our partners, we've released the fiskaly SCU to production. To fulfill the legal requirements, we followed fiskaly's advice and added the informational phrase _"TSE in Evaluierung"_ ("TSE in evaluation") into the newly added certification identification signature item (see below). Please keep in mind that the fiskaly TSE is still in evaluation, and switching to the certified state will require an update of the SCU. Please refer to our [blog post about the state of cloud TSEs](https://fiskaltrust.de/news/aktueller-status-zur-cloud-tse-202103/) for more details.
```

## New signature item: Certification identification
We've added an additional signature item to the receipt response of POS receipts that contains the certification identification of the used TSE (e.g. `BSI-K-TR1234-5677`), and additionally contains a hint about the current state of the certification process or the operational environment. The _ftSignatureType_ of the newly added item is `0x4445000000000022`, more information can be found [here](https://docs.fiskaltrust.cloud/docs/poscreators/middleware-doc/germany/reference-tables/ftsignaturetype).

The signature item was added for all TSEs, including hardware devices. Currently, additional information is displayed for the three cloud SCUs that are available in our production systems:

> | **SCU**                                      | **Certification identification**    |

Reviewer: Should we add the certifications for the other TSEs to that table too?

Author: I'd rather not to highlight the point that those are the "important" ones to print onto the receipt, and added another hint that this is mandatory below

## Compatibility improvement: Fiskal Cloud Connector on Linux
During the last two weeks, our partners informed us that the FCC could not be successfully installed on some Linux distributions without manually making some files executable. We have changed this behavior so that no manual action is needed anymore when setting up a _Swissbit Cloud_ or a _Deutsche Fiskal_ TSE on Linux.

## Bug fix: Deleting logs on fiskaly TSEs
To ensure equal functionality in all our TSE implementations, we've updated our fiskaly SCU to now mark logs that have been exported during a daily closing receipt as deleted. This resolves two issues:
- Previously, the fiskaly SCU always did a full export when daily-closing receipts were processed, and stored the results in the database. When a range-based `/Journal` export of those TAR files was called, the response could hence include more logs than requested. 
- Due to the full export, the duration of daily-closing receipts on fiskaly SCUs increased with each processed transaction, ultimately making the process very slow when many transactions were performed. 

> Summarized, the fiskaly SCU now behaves as our other TSE implementations and only exports TAR logs that were generated since the last daily-closing receipt. However, there's one exception: in case there are open transactions, it's not possible to mark any transactions as deleted; the open transactions need to be closed first.

Reviewer: ```suggestion
Summarized, the fiskaly SCU now behaves as our other TSE implementations and only exports TAR logs that were generated since the last daily-closing receipt. However, there's one exception: similar to the behavior of the hardware TSEs in the case of open transactions, it's not possible to mark any transactions as deleted; the open transactions need to be closed first.
```

## Stability improvement: Add flag to ignore transaction mismatches at daily-closing receipts
In rare cases (e.g. due to power or network failures), TSE transactions could be marked as finished on the TSE, but the Middleware still kept them open in our internal database. As these "open" transactions must be closed before a daily-closing receipt can be executed, this effectively blocked the Middleware from performing this special receipt.

To resolve this problem, we've introduced the _ftReceiptCaseFlag_ `0x0000000020000000` (as described [here](https://docs.fiskaltrust.cloud/docs/poscreators/middleware-doc/germany/reference-tables/ftreceiptcase#ftreceiptcaseflag)). When this flag is sent with the daily-closing receipt, the Middleware will check if transactions that are marked open in our database are actually open in the TSE; if this is the case, a journal entry is made and the database is updated.

> **Please note that this flag should be be used per default, as it may hide power configuration issues or network problems that lead to incomplete DSFinV-K exports**. However, it may be useful in these cases to unblock users of cash registers in case this scenario occurred.

Reviewer: ```suggestion
**Please note that this flag shouldn't be used per default, as it may hide power configuration issues or network problems that lead to incomplete DSFinV-K exports**. However, it may be useful in these cases to unblock users of cash registers in case this scenario occurred.
```
